products/appengine/api.yaml

```diff
@@ -440,3 +440,37 @@ objects:
             name: 'shell'
             description: |
               The format should be a shell command that can be fed to bash -c.
+  - !ruby/object:Api::Resource
+    name: 'ApplicationUrlDispatchRules'
+    description: |
+      Rules to match an HTTP request and dispatch that request to a service.
+    base_url: 'apps/{{project}}'
+    create_url: 'apps/{{project}}?updateMask=dispatch_rules'
+    create_verb: :PATCH
+    delete_verb: :PATCH
+    update_url: 'apps/{{project}}?updateMask=dispatch_rules'
+    update_verb: :PATCH
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      api: 'https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps#UrlDispatchRule'
+    properties:
+      - !ruby/object:Api::Type::Array   
```

Comment:
I expect that all/most of these properties should be `required: true`

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/pull/2466